core/src/main/java/com/scalar/db/api/ScanBuilder.java

```diff
@@ -936,525 +910,299 @@ public Scan build() {
     }
   }
 
-  public static class BuildableScanAllFromExistingWithWhere
-      implements OperationBuilder.Namespace<BuildableScanAllFromExistingWithWhere>,
-          OperationBuilder.Table<BuildableScanAllFromExistingWithWhere>,
-          Consistency<BuildableScanAllFromExistingWithWhere>,
-          Projection<BuildableScanAllFromExistingWithWhere>,
-          Ordering<BuildableScanAllFromExistingWithWhere>,
-          Limit<BuildableScanAllFromExistingWithWhere>,
-          ClearProjections<BuildableScanAllFromExistingWithWhere>,
-          ClearOrderings<BuildableScanAllFromExistingWithWhere>,
-          ClearNamespace<BuildableScanAllFromExistingWithWhere> {
-
-    @Nullable protected String namespaceName;
-    protected String tableName;
-    @Nullable protected final ConditionalExpression condition;
-    protected final Set<Set<ConditionalExpression>> conjunctions = new HashSet<>();
-    protected final Set<Set<ConditionalExpression>> disjunctions = new HashSet<>();
-    protected final List<String> projections = new ArrayList<>();
-    protected final List<Scan.Ordering> orderings = new ArrayList<>();
-    protected int limit;
-    @Nullable protected com.scalar.db.api.Consistency consistency;
+  public static class BuildableScanFromExistingWithWhere
+      implements OperationBuilder.Namespace<BuildableScanFromExistingWithWhere>,
+          OperationBuilder.Table<BuildableScanFromExistingWithWhere>,
+          Consistency<BuildableScanFromExistingWithWhere>,
+          Projection<BuildableScanFromExistingWithWhere>,
+          Ordering<BuildableScanFromExistingWithWhere>,
+          Limit<BuildableScanFromExistingWithWhere>,
+          ClearProjections<BuildableScanFromExistingWithWhere>,
+          ClearOrderings<BuildableScanFromExistingWithWhere>,
+          ClearNamespace<BuildableScanFromExistingWithWhere> {
 
-    private BuildableScanAllFromExistingWithWhere(BuildableScanAllFromExistingWithWhere buildable) {
-      this.namespaceName = buildable.namespaceName;
-      this.tableName = buildable.tableName;
-      this.condition = null;
-      this.conjunctions.addAll(buildable.conjunctions);
-      this.disjunctions.addAll(buildable.disjunctions);
-      this.limit = buildable.limit;
-      this.orderings.addAll(buildable.orderings);
-      this.projections.addAll(buildable.projections);
-      this.consistency = buildable.consistency;
+    BuildableScanOrScanAllFromExisting buildableScanFromExisting;
```

Comment:
```suggestion
    private BuildableScanOrScanAllFromExisting buildableScanFromExisting;
```
This can be `private`?